### PR TITLE
Add missing features to scala3doc to match dotty docs

### PR DESCRIPTION
**Pull request by romanowski**
Links do generated documentation using this PR for [Scala 3](https://scala3doc.virtuslab.com/pr-10320/scala3/index.html) and [scal3doc itself](https://scala3doc.virtuslab.com/pr-10320/self/index.html) (however it has just a token static site).

Scope of this PR:
 - [x] render index page for blog
 - [x] blogposts has all required parameters
 - [x] render images properly in static site/blog
 - [x] search bar is flating
 - [x] Add properties for version, revision etc.
 - [x] edit on github links does not work
 - [x] linking from static pages to types
 - [x] linking to paragraphs
 - [x] ensure that we document all the classes that are documented on https://dotty.epfl.ch/

Known differences:
- navigation is a bit different but allow for adding content of each level (so e.g. https://dotty.epfl.ch/docs/reference/overview.html can be moved directly to `Reference` node: https://scala3doc.virtuslab.com/pr-10320/scala3/docs/Reference/index.html)
- searchbar is dynamic and is able to search in titles of static pages and blogs however we do search for packages (however I am pretty sure @pikinier20 will be able to  make this work)
- For now we do not support includes in static documentation but this is a matter of day or two (documentation for scala 3 uses it on main page and to add a table of conent on one page)
- We do not support (but shoudl not be hard to implement if needed) rendering of docstrings in static doumentation but this is not used in Scala 3 documentation
- Scala3doc doesn not come with predefined layouts. images and includse for static documentation. Once we migrate Scala 3 documentaiton we plan to reach out to community to ask what kind of defaults are expected.
- TODO: inspect other differneces

New features:
 - [site] sidebar can have arbitrary deps, sections (directories) can have its own content (in `index.html` or `index.md`)
 - scala3doc cover all language features (bug in givens is being fixed by @pikinier20)
 - scala3doc has members filter (improved version of one in scaladoc)
 - members in scala3doc is groped by kind and origin (defined and inherited) and in future we can apply arbitraty (also dynamic) groupping
- scala3doc provides subset of implicit members (from extension methods, given and implicit conversions) however this is really a naive implemtnation
- scala3doc provide linear subtypes and type hierarchy diagrams (https://scala3doc.virtuslab.com/pr-10298/scala3/api/scala/-empty-tuple/index.html)
 - scala3doc is fast and reuses compilation artifacts. It takes under 20s seconds on my machine to generate documentation for Scala 3 once code is compiled. Scala3doc was not optimized at any point so I am sure we can be even faster  
- scala3doc has proper implementation of linking to sources that even more powerfull then scaladoc (allows to provide custom patterns as well as partial patterns for subrepositories or documentation compose from various repositories)
- scala3doc has abstract and unified mechanism for linking to pages based on DRIs that will allow us to link to documentation from other projects
- scala3doc properly parsers doc comments (this includes linking etc.)
- scala3doc allow to cusomize logo and has its own custom logo :)

### CODE REVIEW

File: scala3doc/src/dotty/dokka/site/LoadedTemplate.scala [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+import org.jetbrains.dokka.base.renderers.html.{NavigationNode, NavigationPage}
+import org.jetbrains.dokka.model.Documentable
+import org.jetbrains.dokka.pages._
+import org.jetbrains.dokka.transformers.pages.PageTransformer
+import org.jsoup.Jsoup
+import scala.collection.JavaConverters._
+
+
+case class LazyEntry(val getKey: String, value: () => String) extends JMapEntry[String, Object]:
```
Comment (BarkingBad) [inline scala3doc/src/dotty/dokka/site/LoadedTemplate.scala:16]: Redundant val

**Review by BarkingBad — APPROVED** [review]
So far look good

File: scala3doc/src/dotty/dokka/site/common.scala [hunk 2]
```diff
@@ -25,20 +24,19 @@ val apiPageDRI: DRI = mkDRI(packageName = "api", extra = "__api__")
...
-        AnchorLinkExtension.create(),
-        EmojiExtension.create(),
-        YamlFrontMatterExtension.create(),
-        StrikethroughExtension.create()
-      ).asJava)
-    .set(
-      EmojiExtension.ROOT_IMAGE_PATH,
-      "https://github.global.ssl.fastly.net/images/icons/emoji/"
-    )
+    .set(Parser.INDENTED_CODE_BLOCK_PARSER, false)
```
Comment (abgruszecki) [inline scala3doc/src/dotty/dokka/site/common.scala:27]: Do we intentionally disable indented code blocks?

Comment (romanowski) [reply scala3doc/src/dotty/dokka/site/common.scala:27]: I copied the settings over from dotty-doc so I do not have a strong feeling aboout any of this options beside `ANCHORLINKS` ones.

Comment (abgruszecki) [reply scala3doc/src/dotty/dokka/site/common.scala:27]: I think we should support them, unless we have a reason not to.

There's a separate topic here: linking to documentation from static site should be done with Wiki links, like in doc comments. This requires the WikiLink extension like [here](https://github.com/lampepfl/dotty/blob/master/scala3doc/src/dotty/dokka/tasty/comments/MarkdownParser.scala#L33) and some link-handling code that's already in MarkdownConverter [here](https://github.com/lampepfl/dotty/blob/master/scala3doc/src/dotty/dokka/tasty/comments/MarkdownConverter.scala#L87).

**Review by abgruszecki — COMMENTED** [review]
Aside from the comments, LGTM!

**Review by abgruszecki — APPROVED** [review]

**Pull request merged**